### ISSUE TITLE
QSSTV Main Branch Testing Feedback

### ISSUE THREAD

**Issue by vk6mo** [ON4QZ/QSSTV#29]
Environment: LM21.1
QSSTV: CLI build from main branch, labelled as 9.5.11
Done some testing on code.
Main finding is SSTV decodes fine, but DRM (either standard or hybrid) does not decode correctly.
MSC indicator cycles between red, yellow green.
Reverting to 9.5.8 decodes successfully. Back to 9.5.11 does not decode.
Any more info I can provide?
More results of testing to come ...

**Comment by pa2eon** [comment]
Dear Dave,
Can you send me a link or the code from version 9.5.8
I will check it .. and bring it over on QT 6.x

**Comment by vk6mo** [comment]
Hi

 

I don’t know exactly where source for 9.5.8 comes from. 

I got it from LM’s software manager as a prebuilt executable package.

There doesn’t appear to be any labels on the source to indicate at what point 9.5.8 was built.

The best thing I can think of is to go back through the tree at the merge points and see??

What do you reckon (WDYR) ?


Dave

**Comment by vk6mo** [comment]
Hi Eric


Looks like 9.5.8 is available from https://github.com/KD9QZO/qsstv

Version 9.5.8 is labelled.

Tried to attach for you but it was too large for github.

 

 

Dave

 

From: Dave ***@***.***> 
Sent: Wednesday, May 24, 2023 10:42 AM
To: 'ON4QZ/QSSTV' ***@***.***>
Subject: RE: [ON4QZ/QSSTV] QSSTV Main Branch Testing Feedback (Issue #29)

 

Hi

 

I don’t know exactly where source for 9.5.8 comes from. 

I got it from LM’s software manager as a prebuilt executable package.

There doesn’t appear to be any labels on the source to indicate at what point 9.5.8 was built.

The best thing I can think of is to go back through the tree at the merge points and see??

What do you reckon (WDYR) ?


Dave

**Comment by pa2eon** [comment]
Dear All,
I have rewritten the 'old' code version 9.5.8 to use within the QT6 creator (Vs. 6.2.4 ).
But the analogue SSTV signals will decode but the DRM signals are strong but not decoded correct.
The 'Time', 'Frame', 'FAC' will be green most, but the 'MSC'  is most of time yellow.

Any idea where to find this 'bug'  in the code ?

vy' 73, Eric

**Comment by vk6mo** [comment]
Hi

 

How did you get on with this?

 

Do you want me to look into this at all or just wait for you to resolve? 

I can set up an older working scenario (9.5.8) as well as current scenario both under Creator so test side by side if you think that would be helpful.

 

Currently, I’m just looking through various parts of the code, trying to get an understanding of how it all works together.

 

I have a number of ideas for improving code, but getting current code working under Qt6 seems obvious priority to me.

 

Regards

 


Dave

**Comment by dl8dtl** [comment]
Well, if you can spot any possible reason, that would be great.
Is it already sure that it's the decoder that fails? I.e., passing the same audio sample into the old version works, while it fails in the current code?

**Comment by pa2eon** [comment]
Is it possible that I can get the same audio sample .. to find out the problem ..

You can use 'wetransfer' to send it to me ..

Met vriendelijke groeten,
Kind regards,

Eric Oosterbaan / PA2EON
T: +31 73 5530052
E: ***@***.***
H: http://www.pa2eon.nl
W: http://www.amsat-nl.org

----- Originele bericht -----
Van: Jörg Wunsch ***@***.***)
Datum: 30-05-2023 08:22
Naar: ON4QZ/QSSTV ***@***.***)
Cc: Eric ***@***.***), Comment ***@***.***)
Onderwerp: Re: [ON4QZ/QSSTV] QSSTV Main Branch Testing Feedback (Issue #29)

Well, if you can spot any possible reason, that would be great.
Is it already sure that it's the decoder that fails? I.e., passing the same audio sample into the old version works, while it fails in the current code?

**Comment by vk6mo** [comment]
“Is it already sure that it's the decoder that fails?”

Not definitely. 

 

Same audio stream to EasyPal and Q9.5.8 both repeatedly work.

Same audio stream to Easypal and QSSTV  latest version (9.5.11 or later), QSSTV repeatedly fails (EasyPal is fine).

No change to settings or audio level between two different QSSTV versions.

If EasyPal fails or is marginal, I have disregarded pass/fail or QSSTV.

Not one successful DRM decode on new QSSTV.

15-20 successful DRM decodes on current QSSTV per day (one day its hybrid, one day its standard).

 

Jorg, do you think it could be something else? Is there somewhere else you think we should look?

 

PS: SSTV decodes fine.

 

I’m happy to give it a go. Because of lack of familiarity with code, might take a while. 

But if you are already finding the problem, no sense duplicating work….??

 


Dave

**Comment by vk6mo** [comment]
> Is it possible that I can get the same audio sample .. to find out the problem ..

I have two audio samples (two pictures sent via DRM standard mode). Whoever wants a copy, please let me know your email. I will transfer via pCloud (no signup for me or you). Files are approx 20MB, 40MB. You will receive an email with link to download.

**Comment by pa2eon** [comment]
yep .. ***@***.***,Met vriendelijke groeten,Eric OosterbaanM: ***@***.***: www.pa2eon.nlW2: www.amsat-nl.org
-------- Oorspronkelijk bericht --------Van: Dave ***@***.***> Datum: 03-06-2023  11:44  (GMT+01:00) Aan: ON4QZ/QSSTV ***@***.***> Cc: Eric ***@***.***>, Comment ***@***.***> Onderwerp: Re: [ON4QZ/QSSTV] QSSTV Main Branch Testing Feedback (Issue #29) 

Is it possible that I can get the same audio sample .. to find out the problem ..

I have two audio samples (two pictures sent via DRM standard mode). Whoever wants a copy, please let me know your email. I will transfer via pCloud (no signup for me or you). Files are approx 20MB, 40MB. You will receive an email with link to download.

**Comment by vk6mo** [comment]
Looks like GitHub removed your email address .. 
From your website it appears that it is pa2eon at amsat dot org 
(spelt out so hopefully github won't edit). Correct?
